### PR TITLE
tests: add a test for calling API methods w/o calling `configure()`

### PR DESCRIPTION
**Pull request by imaginaris**
# Description
Added a test to check if module won't crash if `configure()` is not called

# Checklist
- [X] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [X] I wrote/updated tests for new/changed code
- [X] I removed all sensitive data **before every commit**, including API endpoints and keys
- [X] I ran `fastlane ci` without errors

### CODE REVIEW

File: Tests/PublicAPISpec.swift [hunk 1]
```diff
@@ -72,6 +72,19 @@ class PublicAPISpec: QuickSpec {
 
         describe("RInAppMessaging") {
 
+            it("will not crash if api methods are called prior to configure()") {
+                RInAppMessaging.deinitializeModule()
+                expect(RInAppMessaging.initializedModule).to(beNil())
+
+                let errorDelegate = ErrorDelegate()
+                RInAppMessaging.delegate = delegate
+                RInAppMessaging.errorDelegate = errorDelegate
```
Comment (donnie-jp) [inline Tests/PublicAPISpec.swift:81]: Should we call the error delegate if an API is call without the module being initialized?

**Review by donnie-jp — APPROVED** [review]
LGTM

File: Tests/PublicAPISpec.swift (hunk 1 above)
Comment (imaginaris) [reply Tests/PublicAPISpec.swift:81]: Good idea. Done.

File: Tests/PublicAPISpec.swift [hunk 2]
```diff
@@ -72,6 +72,34 @@ class PublicAPISpec: QuickSpec {
...
+
+                let errorDelegate = ErrorDelegate()
+                RInAppMessaging.errorDelegate = errorDelegate
+
+                RInAppMessaging.delegate = delegate
+                RInAppMessaging.accessibilityCompatibleDisplay = true
+                RInAppMessaging.closeMessage(clearQueuedCampaigns: true)
+                RInAppMessaging.logEvent(LoginSuccessfulEvent())
+                RInAppMessaging.registerPreference(IAMPreferenceBuilder().setUserId("user").build())
+                expect(errorDelegate.totalErrorNumber).toEventually(equal(5))
```
Comment (imaginaris) [inline Tests/PublicAPISpec.swift:100]: I'm not sure if it's better to check error description after each call.

Comment (donnie-jp) [reply Tests/PublicAPISpec.swift:100]: I think this is sufficient but it'd maybe be helpful to have a `// 1st error sent`, `// 2nd` etc. for each API call to make it really obvious where the 5 comes from.

If you want to check the error desc too I'd have a separate test for each API call.

**Review by donnie-jp — APPROVED** [review]

**Pull request merged**